### DISCUSSION TITLE
Can the links function load data dynamically, like the metas function?

### DISCUSSION THREAD

**Discussion by kmvan** [remix-run/react-router#12534] [Q&A] [answered]
```typescript
export const links = (/* not allow param */) => {
// I'd want to get server data.
...
}
```
Any idea?

**Comment by sergiodxa** [comment, marked as answer]
No, links function receive no parameter, this is done this way to links can be preloaded before a navigation when using `<Link preload>`, otherwise RR would have to first fetch the data and then preload links.

If you need to add links based on data from your loaders, you can use the MetaFunction which can return links too.

**Reply by kmvan** [reply]
Got it. Thanks man.
